main.tf

```diff
@@ -1,28 +1,31 @@
 data "aws_caller_identity" "current" {}
 
-data "aws_sns_topic" "bucket-arn" {
-  name = "bucket-arn"
-}
+resource "aws_s3_bucket_notification" "bucket_notification" {
+  bucket = aws_s3_bucket.default.id
 
+  topic {
+    topic_arn     = var.sns_arn
+    events        = [var.event]
```

Comment:
rather than passing a string, pass a list

Reply:
This has also been corrected.